### ISSUE TITLE
Use Trello 

### ISSUE THREAD

**Issue by sagotsky** [openscholar/integrated_support#96]
Integrated support still isn't quite up to snuff yet.  Let's throw more APIs at it.  Instead of webhooks, labels, and voodoo... 
# Use Trello to manage the roadmap.

We'll have one intake board with bugs and ideas imported from GetSatisfaction and Engineering from Github:

![screenshot](http://imgur.com/XsdwWOJ.png)

Nothing on this board is set in stone yet.  It's just a collection of the ideas we've received.  

We will also have a roadmap board with a backlog list (aka the In the Future milestone) and a list for each planned milestone:

| Backlog | Sept 16 | Oct 30 | Dec 14 |
| --- | --- | --- | --- |
| etc | etc | etc | etc |

This board does indicate what's planned and will be used to generate the roadmap page.  The /roadmap page will just be a rendering of what's on these lists.  /backlog will be similar, but for the backlog list.  The idea is that the roadmap should display items from exactly one easily managed source, instead of patching it together from a variety of sources and allowing a number of events and labels to affect it.
# Use Trello events for notifications

All event notifications should trickle down from trello into our other services.  
- [x] Ticket creation - 
- [ ] ~~Scheduling notification~~ (GS and desk will be notified by github events, not by trello events)
- [ ] ~~Completion~~  (GS and Desk will be notifyed by GH events)
- [x] Milestone changes. - When a trello card moves in the milestone board, github should be updated.
- [x] User assignment.  When a card is assigned to someone, assign it to their GH account.  (what do if multiple trello assignees?)
## 

**TODO 09/11/13**
- [x] Fix labels.  They're not applying on osops.
- [x] Get GH status via ajax
- [x] Never include PRs.  Only issues.
- [x] Github should only use items tagged with engineering.
- [x] GS topics should include links to GS and GH if both are available.
- [x] GH issues should be excluded if they are mapped.  Let the GS topic have both.

**TODO 09/12/13**
- [x] Make it purty.  Dan's labels are sweet.
- [x] New design:
  1. Current mielstone, rendered.
  2. ~~List of links to each upcoming milestone.~~
  3. Backlog, rendered as three columns.
  4. List of links to past milestones.
  5. ~~Links load table with ajax.  Each includes a permalink to just that table.~~
- ~~Find out why trello missed http://community.openscholar.harvard.edu/openscholar/topics/allow_site_owners_to_redirect_people_profile_pages~~
- [x] When a GS topic enters a milestone list, if it isn't linked yet create the GH issue and link them.  
- [x] GS topics can get their status from linked GH issues
- [x] Pretty urls.  urlencode gives us `Sept%2016`.  Replacing spaces with dashes would be nicer.
## 

**TODO 09/18/13**
- [x] No ajax.  Just use regular pages for simplicity's sake.
- [x] All navigation should take place in the select box.  No more tabs.
- [x] Legend w/ checkboxes.  Show/hide different statuses.
- [x] ~~Double check that labels are being created.  (Maybe check on page loads too.  If a case is loaded with no label, maybe we can add it retroactively?)~~ (Labels were working fine.  What wasn't was Trello on osops.  With that out of commission labels couldn't possibly update).
## 

**TODO 09/19/13**
- [x] Checkboxes should opt in like taxonomies.  Clicking done shows only done.  Adding in testing shows done and testing.  All are off by default, so nothing is shown.
- [x] Checkboxes to bottom
- [x] Span around label text
- [x] Fourth column for hotfixes.  Populate from GH.
- [x] Display on an openscholar themed vsite, so @dancabral's 4th column will flex. 
- [x] Call it roadmap2.  The trello part is a secret.
- [x] Backlog is a separate link (always visible), not part of select.
- ~~Fix select for viewing within a vsite.~~ (This will be appearing on a vsite with a domain, so it's no problem)
## 

**TODO 09/20/13**
- [x] **Releases/Backlog as tabs**
- [x] 'select a milestone' -> 'select a release'
- [x] No release archive (for now)
- [x] Order: Features, Bugs, Engineering
- [x] Select and title are redundant.  Remove the title and select the current milestone.

Backlog
- [x] No hotfixes
- [x] No status
- [x] First tab is projects

When there's time...
- [ ] Filter widget inspired by trello label selector.  
  - Filters are off by default (but colored)
  - When all are off, no filtering
- [ ] Import tickets from previous milestones.  Probably loop over one milestone in each pass.  Can they all be engineering, or should we use GS for those where that works.  Alternatively, we could use type: feature/bug...
- [ ] Add a scrollbar lib?  I'm partial to jquery's perfect-scrollbar plugin.
